install_files/ansible-base/roles/restore/tasks/perform_restore.yml

```diff
@@ -53,11 +53,40 @@
       - "More info: {{ compare_result.stdout }}"
   when: not restore_skip_tor
 
+- name: Calculate local backup file checksum
+  connection: local
+  become: no
+  stat:
+    path: "{{ restore_file }}"
+    checksum_algorithm: sha256
+  register: local_backup_file
+  when: restore_manual_transfer
+
+- name: Calculate remote backup file checksum
+  stat:
+    path: "/tmp/{{ restore_file }}"
```

Comment:
What do you think about allowing a full path, to allow for `/tmp` on the server not having enough free space?

Reply:
I'm not in favour of it - points against:
-The SD setup instructions as they stand don't result in a separate `/tmp` partition, so if there's enough space on `/` this isn't an issue - and the backup job would have failed in the first place as it also uses `/tmp`
- this is the current backup location - it would introduce more changes and would require extra work to make sure backup tempfiles are purged from whatever alternate location was specified.

Reply:
In general, writing sensitive files to `/tmp` is not recommended [1], due to the permissions of that partition, and the ability of all system users to write to it. In this case, the servers are effectively single-user systems, so the risks are relatively low, but in the case where an attacker has unprivileged access to the system, they may have access to more information.

[1] some links/references in https://security.openstack.org/guidelines/dg_using-temporary-files-securely.html